processor/attributesprocessor/attributes.go

```diff
@@ -122,6 +122,10 @@ func (a *attributesProcessor) ConsumeTraceData(ctx context.Context, td consumerd
 	return a.nextConsumer.ConsumeTraceData(ctx, td)
 }
 
+func (a *attributesProcessor) GetCapabilities() processor.Capabilities {
```

Comment:
total nit and out of scope for this pr: how do we ensure that processors that mark themselves as false truly don't mutate the data?

Reply:
We don't. Processors declare the intent, we don't enforce it at runtime in any way. Obviously this has to be an enforcement topic during code reviews.